app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt

```diff
@@ -35,7 +41,23 @@ class WireNotificationManager @Inject constructor(
     }
 
     private suspend fun fetchAndShowCallNotificationsOnce(userId: QualifiedID) {
-        //TODO
+        //TODO for now GetIncomingCallsUseCase() returns valid data not from the first try.
+        // so it's possible to have scenario, when FCM comes informing us that there is a Call,
+        // but we don't get it from the first GetIncomingCallsUseCase() call.
+        // To cover that case we have this `intervalFlow().take(6)`
+        // to try get incoming calls 6 times, if it returns nothing we assume there is no incoming call
+        intervalFlow(CHECK_INCOMING_CALLS_PERIOD_MS)
+            .map {
+                coreLogic.getSessionScope(userId)
+                    .calls
+                    .getIncomingCalls()
+                    .first()
```

Comment:
On multiple calls in parallel we will only see the oldest one with `first()`. Is that intended behavior?

Reply:
i assume that comment should be not here, but in `CallNotificationManager` (as this is `Flow.first()`) :) 

But yes, taking the oldest one makes sense IMO: the first caller is prioritized.